pkg/kubelet/kubelet.go

```diff
@@ -2577,10 +2577,12 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 				podCopy := pod.DeepCopy()
 				kl.updateContainerResourceAllocation(podCopy)
 
-				// Check if we can admit the pod; if not, reject it.
-				if ok, reason, message := kl.canAdmitPod(activePods, podCopy); !ok {
-					kl.rejectPod(pod, reason, message)
-					continue
+				if pod.Status.Phase == v1.PodPending && pod.Status.PodIP == "" {
```

Comment:
Could you please explain the rationale for this check and why is it sufficient, preferably in the commit message?
I'm not sure this guard is complete enough

Reply:
The admission check for kubelet should be a one-time thing. For pods that have been admitted, no matter how the node configuration changes, it should not affect the pods that have been admitted. There are two main points to check here. One is whether there is an IP. If the IP has been assigned, it proves that the pod already has a sandbox and must have been admitted. The other is whether the pod is in the pending state. The pod in the non-pending state must have been processed by kubelet, because from pending to non- The conversion of Pending state is done by kubelet.